internal/controllers/github.go

```diff
@@ -155,6 +160,34 @@ func (g *GitHub) ValidateWebhookSecret(secret []byte, headers map[string]string)
 	return g.WebhookSecret.ValidateSignature(secret, headers)
 }
 
+// PromotionTargetRefExists checks if a ref exists in the repository
+func (g *GitHub) PromotionTargetRefExists(ctx *promotion.Context) bool {
+	_, _, err := ctx.ClientV3.Git.GetRef(g.ctx, *ctx.Owner, *ctx.Repository, helpers.NormaliseFullRef(ctx.BaseRef))
+	return err == nil
+}
+
+// CreatePromotionTargetRef creates a new ref in the repository
+func (g *GitHub) CreatePromotionTargetRef(ctx *promotion.Context) (*github.Reference, error) {
+	// Fetch the first commit on the head ref
+	commits, _, err := ctx.ClientV3.Repositories.ListCommits(g.ctx, *ctx.Owner, *ctx.Repository, &github.CommitsListOptions{
+		SHA: *ctx.HeadRef,
+	})
```

Comment:
I fear this may be sub-optimal: as the promotion repository's history grows, this necessarily grows to an ever-increasing number of API calls (in the best case scenario that it's automatically iterating through all pages of results). This could be a great opportunity to use the GraphQL API via `ctx.ClientV4` to explicitly get just the *first* commit (cf. https://docs.github.com/en/graphql/reference/objects#commit)

Reply:
Unfortunately this doesn't seem possible with the current GraphQL API as commits are sorted from the HEAD in descending order. I tried:
```graphql
{
  repository(owner: "<owner>", name: "<repo>") {
    ref(qualifiedName: "<branch>") {
      target {
        ... on Commit {
          history(last: 1) {
            edges {
              node {
                oid
              }
            }
          }
        }
      }
    }
  }
}
```
But it seems that the API needs a cursor and is expecting to paginate from HEAD until it reaches the root commit... :-(

Reply:
^ Added pagination.